### DISCUSSION TITLE
Streaming SSR

### DISCUSSION THREAD

**Discussion by tigawanna** [rakkasjs/rakkasjs#103] [Q&A] [answered]
Was just wondering how streaming worked in rakkas and if it works when you wrap a component in client suspense or is automatically taken care of

**Comment by cyco130** [comment, marked as answer]
It's automatically taken care of.

Rakkas wraps your entire app in Suspense (with an empty fallback) but you can add further Suspense boundaries for more granular loading state. React will then render the fallback and replace it with the actual content once it's ready.

Note that streaming is disabled under certain circumstances:

- When the request is from a bot (for SEO purposes)
- When prerendering statically
- On Vercel and Netlify serverless environments (AWS which they're based on doesn't support streaming responses)
- When turned off by you (by setting `throttleRenderStream` to true in a [`headers` function](https://rakkasjs.org/guide/status-and-headers)) e.g. if you want to use a lib that doesn't support streaming like Styled Components.

If it's disabled, Rakkas will wait for the whole page to render before sending the HTML.

**Comment by tigawanna** [comment]
Thank you for the informative answer
